internal/configs/virtualserver_test.go

```diff
@@ -894,12 +929,109 @@ func TestGenerateUpstreamForZeroEndpoints(t *testing.T) {
 		Servers: nil,
 	}
 
-	result = generateUpstream(name, upstream, endpoints, isPlus, &cfgParams)
+	result = generateUpstream(name, upstream, vsEx, isPlus, isResolvedConfigured, &cfgParams)
 	if !reflect.DeepEqual(result, expectedForNGINXPlus) {
 		t.Errorf("generateUpstream(isPlus=%v) returned %v but expected %v", isPlus, result, expectedForNGINXPlus)
 	}
 }
 
+func TestGenerateUpstreamForExternalNameService(t *testing.T) {
+	name := "test-upstream"
+	vsEx := &VirtualServerEx{
+		VirtualServer: &conf_v1alpha1.VirtualServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      name,
+				Namespace: "test-namespace",
+			},
+		},
+		Endpoints: map[string][]string{
+			"test-namespace/test-upstream:0": []string{"example.com"},
+		},
+		ExternalNameSvcs: map[string]bool{
+			name: true,
+		},
+	}
+	cfgParams := ConfigParams{
```

Comment:
these cfgParams values doesn't seem to be related to the external name svc at all

Reply:
👍 from copy/paste